proto/umee/incentive/v1/tx.proto

```diff
@@ -0,0 +1,111 @@
+  // In the current implementation, the sponsor must be a single account and the MsgSponsor must fully cover the expected program rewards.
+  rpc Sponsor(MsgSponsor) returns (MsgSponsorResponse);
+
+  // GovSetParams is used by governance proposals to update parameters.
+  rpc GovSetParams(MsgGovSetParams) returns (MsgGovSetParamsResponse);
+
+  // GovCreateProgram is used by governance proposals to create (but not fund) an incentive program.
+  rpc GovCreateProgram(MsgGovCreateProgram) returns (MsgGovCreateProgramResponse);
+}
+
+// MsgClaim represents a account's request to claim pending rewards.
+message MsgClaim {
+  string account = 1;
+}
+
+// MsgClaimResponse defines the Msg/Claim response type.
+message MsgClaimResponse {
+  repeated cosmos.base.v1beta1.Coin amount = 1 [
+    (gogoproto.nullable)     = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+}
+
+// MsgBond represents a account's request to bond uToken collateral.
+message MsgBond {
+  string                   account = 1;
+  uint32                   tier    = 2;
+  cosmos.base.v1beta1.Coin asset   = 3 [(gogoproto.nullable) = false];
+}
+
+// MsgBondResponse defines the Msg/Lock response type.
+message MsgBondResponse {}
+
+// MsgBeginUnbonding represents a account's request to begin unbonding uToken collateral.
+message MsgBeginUnbonding {
+  string                   account = 1;
+  uint32                   tier    = 2;
+  cosmos.base.v1beta1.Coin asset   = 3 [(gogoproto.nullable) = false];
+}
+
+// MsgBeginUnbondingResponse defines the Msg/BeginUnbonding response type.
+message MsgBeginUnbondingResponse {}
+
+// MsgSponsor represents a sponsor's request to fund rewards for an incentive program.
+// The program must have been passed by governance, not yet started, and not yet funded.
+// Funded assets must be the full amount required by the program.
+message MsgSponsor {
+  // Sponsor bech32 account address
+  string                   sponsor = 1;
+  uint32                   program = 2;
+  cosmos.base.v1beta1.Coin asset   = 3 [(gogoproto.nullable) = false];
+}
+
+// MsgSponsorResponse defines the Msg/Sponsor response type.
+message MsgSponsorResponse {}
+
+// MsgGovSetParams is used by governance to update module parameters.
+message MsgGovSetParams {
+  option (gogoproto.equal)           = true;
+  option (gogoproto.goproto_getters) = false;
+  option (cosmos.msg.v1.signer)      = "authority";
+
+  // authority is the address of the governance account.
+  string authority   = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  string title       = 2;
+  string description = 3;
+  Params params      = 4 [(gogoproto.nullable) = false];
+}
+
+// MsgGovSetParamsResponse defines the Msg/SetParams response type.
+message MsgGovSetParamsResponse {}
+
+// MsgGovCreateProgram is used by governance to create an incentive program.
+message MsgGovCreateProgram {
```

Comment:
let's add a comment to what happens with a program if nobod will sponsor it. I suggest to remove it after some time, eg one month.

Reply:
Current plan is to cancel it (move to completed programs with funded amount = zero) at its exact intended start time

This would also happen if the program contained a token that had since been blacklisted or was never listed in the leverage module to begin with (e.g. ibc denom typo).